build/common.go

```diff
@@ -95,6 +96,10 @@ func buildBackend(cfg Config) error {
 	}
 
 	info := getBuildInfoFromEnvironment()
+	pluginID, err := internal.GetStringValueFromJSON(filepath.Join("src", "plugin.json"), "id")
```

Comment:
see `getExecutableName` which already reads the plugin JSON. You can modify that function to return the plugin ID as well (or use the same approach).

This won't take into account if the plugin.json is in a different folder than `src`.

Reply:
> This won't take into account if the plugin.json is in a different folder than src.

Can I just do this instead rather than edit `GetExecutableFromPluginJSON`?

```suggestion
	pluginID, err := internal.GetStringValueFromJSON(pluginJSONPath, "id")
```

This is the same path we use for `getExecutableName`:

https://github.com/grafana/grafana-plugin-sdk-go/blob/72be334272f29f2420ec338b27a441e858520660/build/common.go#L70

Reply:
yeah, it's the same since we don't need to look in inner folders like for the executable name